# <font color=red> _Compte rendu du mini projet Pierre Feuille Ciseaux_ </font>
## <font color=blue> _Morjene Nouha_ </font>
### _2LM1 TP2_

# <font color=green> _Simulation du jeux Pierre Feuille Ciseaux_ </font>

<img src="téléchargement.png">

### <font color=blue>_Principe:_ </font>
    - le jeu se joue en duel, le score maximal de l'un des joueurs est n 
    - Une fois à trois les joueurs révèlent leur main (pierre, feuille ou ciseaux) en même temps: La plus forte des formes l’emporte et le joueur marque le point gagnant
    - Si les deux joueurs utilisent la même forme c’est un match nul

### <font color=blue>_Les règles du jeu:_ </font>
     - la pierre écrasse les ciseaux et gagne
     - La feuille enveloppe la pierre et gagne
     - les ciseaux découpent la feuille et gagnent

#### <font color=red>_Objectifs:_ </font>
      - Objectif1: Simuler le jeu
      - Objectif2: déterminer la probablité de gagner dans ce jeu

### <font color=red> Simulation du choix : </font>

In [57]:
import numpy as np

In [58]:
def choix():
    return int(np.random.choice(range(1,4),1))
#avoir une valeur de l'ensemble{1,2,3} aléatoirement et on transforme le tableau par la fonction int() en un entier

In [59]:
choix()

2

In [60]:
def description(m):
        if m==1:
            return "pierre"
        if m==2:
            return "feuille"
        if m==3:
            return "ciseaux"


In [61]:
description(2)

'feuille'

- 1 désigne la pierre
- 2 désigne la feuille
- 3 désigne les ciseaux


### <font color=red> Simulation du jeux:

In [77]:
def Sim(n):
    j1=0 #choix du joueur1 
    j2=0 #choix du joueur2
    score1=0 #score du joueur1
    score2=0 #score du joueur2
    while score1<=n and score2<=n:
        j1=choix()
        j2=choix()
        print(j1,j2)
        if j1==1:
            if j2==1:
                print('le match est nul')
                print('le score du joueur1:',score1)
                print('le score du joueur2:',score2)
            if j2==2:
                score2=score2+1
                print('le gagnant est',description(j2))
                print('le score du joueur1:',score1)                
                print('le score du joueur2:',score2)
            if j2==3:
                score1=score1+1
                print('le gagnant est:',description(j1))
                print('le score du joueur1:',score1)                
                print('le score du joueur2:',score2)
        if j1==2:
            if j2==1:
                score1=score1+1
                print('le gagnant est:',description(j1))
                print('le score du joueur1:',score1)                
                print('le score du joueur2:',score2)
            if j2==2:
                print('le match est nul')
                print('le score du joueur1:',score1)                
                print('le score du joueur2:',score2)
            if j2==3:
                score2=score2+1
                print('le gagnant est:',description(j2))
                print('le score du joueur1:',score1)                
                print('le score du joueur2:',score2)
        if j1==3:
            if j2==1:
                score2=score2+1
                print('le gagnant est:',description(j2))
                print('le score du joueur1:',score1)                
                print('le score du joueur2:',score2)
            if j2==2:
                score1=score1+1
                print('le gagnant est:',description(j1))
                print('le score du joueur1:',score1)                
                print('le score du joueur2:',score2)
            if j2==3:
                print('le match est nul')
                print('le score du joueur1:',score1)                
                print('le score du joueur2:',score2)
    if score1>score2:
         return "joueur1"
    else:
         return "joueur2"

In [79]:
Sim(5)

3 2
le gagnant est: ciseaux
le score du joueur1: 1
le score du joueur2: 0
3 3
le match est nul
le score du joueur1: 1
le score du joueur2: 0
3 1
le gagnant est: pierre
le score du joueur1: 1
le score du joueur2: 1
3 3
le match est nul
le score du joueur1: 1
le score du joueur2: 1
1 3
le gagnant est: pierre
le score du joueur1: 2
le score du joueur2: 1
3 2
le gagnant est: ciseaux
le score du joueur1: 3
le score du joueur2: 1
3 2
le gagnant est: ciseaux
le score du joueur1: 4
le score du joueur2: 1
2 3
le gagnant est: ciseaux
le score du joueur1: 4
le score du joueur2: 2
1 2
le gagnant est feuille
le score du joueur1: 4
le score du joueur2: 3
1 3
le gagnant est: pierre
le score du joueur1: 5
le score du joueur2: 3
1 2
le gagnant est feuille
le score du joueur1: 5
le score du joueur2: 4
1 2
le gagnant est feuille
le score du joueur1: 5
le score du joueur2: 5
2 2
le match est nul
le score du joueur1: 5
le score du joueur2: 5
1 3
le gagnant est: pierre
le score du joueur1: 6
le score du jou

'joueur1'

In [80]:
#garder le résultat dans une liste L:
L=[]
L.append(Sim(5))
print(L)

1 2
le gagnant est feuille
le score du joueur1: 0
le score du joueur2: 1
1 1
le match est nul
le score du joueur1: 0
le score du joueur2: 1
2 3
le gagnant est: ciseaux
le score du joueur1: 0
le score du joueur2: 2
1 1
le match est nul
le score du joueur1: 0
le score du joueur2: 2
3 3
le match est nul
le score du joueur1: 0
le score du joueur2: 2
3 1
le gagnant est: pierre
le score du joueur1: 0
le score du joueur2: 3
1 1
le match est nul
le score du joueur1: 0
le score du joueur2: 3
1 1
le match est nul
le score du joueur1: 0
le score du joueur2: 3
3 3
le match est nul
le score du joueur1: 0
le score du joueur2: 3
2 2
le match est nul
le score du joueur1: 0
le score du joueur2: 3
2 2
le match est nul
le score du joueur1: 0
le score du joueur2: 3
1 2
le gagnant est feuille
le score du joueur1: 0
le score du joueur2: 4
1 3
le gagnant est: pierre
le score du joueur1: 1
le score du joueur2: 4
3 2
le gagnant est: ciseaux
le score du joueur1: 2
le score du joueur2: 4
1 2
le gagnant est feuil

### <font color=red> Simulation de $10^2$  scénario : </font>

In [88]:
V=10**2
L=[]
for i in range(V):
    L.append(Sim(5))
print(L)

1 3
le gagnant est: pierre
le score du joueur1: 1
le score du joueur2: 0
3 3
le match est nul
le score du joueur1: 1
le score du joueur2: 0
3 1
le gagnant est: pierre
le score du joueur1: 1
le score du joueur2: 1
1 3
le gagnant est: pierre
le score du joueur1: 2
le score du joueur2: 1
2 2
le match est nul
le score du joueur1: 2
le score du joueur2: 1
2 2
le match est nul
le score du joueur1: 2
le score du joueur2: 1
2 1
le gagnant est: feuille
le score du joueur1: 3
le score du joueur2: 1
3 2
le gagnant est: ciseaux
le score du joueur1: 4
le score du joueur2: 1
2 3
le gagnant est: ciseaux
le score du joueur1: 4
le score du joueur2: 2
3 1
le gagnant est: pierre
le score du joueur1: 4
le score du joueur2: 3
2 2
le match est nul
le score du joueur1: 4
le score du joueur2: 3
1 2
le gagnant est feuille
le score du joueur1: 4
le score du joueur2: 4
3 1
le gagnant est: pierre
le score du joueur1: 4
le score du joueur2: 5
2 3
le gagnant est: ciseaux
le score du joueur1: 4
le score du joueur2: 

In [89]:
L.count('joueur1') #compter le nombre d'occurence du joueur 1 dans la liste(le nombre de gains du joueur1)"

51

In [90]:
def Freq(L):
    return L.count('joueur1')/len(L)
Freq(L)

0.51

## <font color=red> Conclusion: </font> 
<b> En jouant $10^2$ fois, on remarque que la probabilité de gagner dans ce jeu est égale à 0.51
   
  